### ISSUE TITLE
SDL2: display.mode ignores depth value

### ISSUE THREAD

**Issue by charlesej** [pygame/pygame#991]
The depth value is ignored when setting the display mode for pygame SDL2 builds. The depth can be set correctly for pygame SDL1.2 builds.

Code example to show the depth value being ignored.
```python
import pygame

pygame.display.init()

for depth in (8, 16, 24, 32):
    surface = pygame.display.set_mode((200, 200), 0, depth)
    print('set depth = {}, get depth = {}'.format(
        depth, surface.get_bitsize()))

pygame.display.quit()
```

Output from the above code.
```
pygame 2.0.0.dev0 (SDL 2.0.9, python 3.7.2)
Hello from the pygame community. https://www.pygame.org/contribute.html
set depth = 8, get depth = 32
set depth = 16, get depth = 32
set depth = 24, get depth = 32
set depth = 32, get depth = 32
```

System details:
- os: windows 10 (64bit)
- python: 3.7.2 (64bit)
- pygame: 2.0.0.dev0 (SDL: 2.0.9) at 0f051ad9423350a215a43ac72f1c98581f22579e

**Comment by robertpfeiffer** [comment]
Old Pygame also ignored some flags and depth parameters. Some were more like *requests*, and you had to check whether you actually got the right kind of display surface.

**Comment by illume** [comment]
Yeah, depth is requests. However, SDL2 definitely supports more than this depth.

I can reproduce it on MacOS 10.13.6
```
$ python display_bpp.py 
pygame 2.0.0.dev5 (SDL 2.0.10, python 2.7.16)
Hello from the pygame community. https://www.pygame.org/contribute.html
set depth = 8, get depth = 32
set depth = 16, get depth = 32
set depth = 24, get depth = 32
set depth = 32, get depth = 32
```

**Comment by robertpfeiffer** [comment]
I can implement this to work as expected, but I'd think most of the time you actually don't want this. I know old PGS4A had always used 16bpp depth display surfaces because the of GLES 1.0 it used under the hood, but barring hardware restrictions, there is no good reason to use an 8-bit indexed display surface with SDL2. And there is certainly no good reason to use a 16-bit surface.

The only reason to support this is backward compatibility.

**Comment by ThanhChinhBK** [comment]
I want to investigate on this issue, is it OK?

**Comment by ankith26** [comment]
Feel free to go ahead, but I suspect it won't be easy (considering SDL API doesn't expose this anymore I think).
If this needs elaborate and complex changes that significantly increases the maintenance burden, it may potentially not be worth it. That being said if the change ends up being small, it would be a very much welcome and appreciated addition!
